src/CMakeLists.txt

```diff
@@ -10,13 +10,19 @@ add_library(secp256k1_precomputed OBJECT EXCLUDE_FROM_ALL
 # from being exported.
 add_library(secp256k1 secp256k1.c $<TARGET_OBJECTS:secp256k1_precomputed>)
 
+# Create a helper lib that parent projects can use to link secp256k1 into a
+# static lib.
+add_library(secp256k1_objs INTERFACE EXCLUDE_FROM_ALL)
```

Comment:
This command fails with CMake 3.16:
```
CMake Error at src/CMakeLists.txt:15 (add_library):
  add_library INTERFACE library may not be used with EXCLUDE_FROM_ALL.
```

This [signature](https://cmake.org/cmake/help/latest/command/add_library.html#interface-with-sources) has been added in CMake 3.19.

Rebase on top of https://github.com/bitcoin-core/secp256k1/pull/1675?